lstchain/calib/camera/sampling_interval_coefficient_calculate.py

```diff
@@ -0,0 +1,170 @@
+        r1_sample_start = r0_r1_calibrator.r1_sample_start.tel[tel_id]
+
+        # Check pulse
+        pulse_event_flag = np.sum(np.max(waveform[gain], axis=1) > 100) > 1800
+    
+        if pulse_event_flag:
+
+            # find pulse peak position
+            pulse_peak = np.argmax(waveform[gain], axis=1)
+            fc = first_capacitors[tel_id][gain]
+            pulse_peak_abs_pos = (fc + r1_sample_start + pulse_peak) % N_CAPACITORS_CHANNEL
+            
+            # increment peak count array
+            self.peak_count[np.arange(N_PIXELS), pulse_peak_abs_pos] += 1
+            self.fc_count[np.arange(N_PIXELS), fc % N_CAPACITORS_CHANNEL] += 1
+            
+
+    def stuck_single_sampling_interval(self, file_list, gain):
+
+        for single_file in file_list:
+
+            run_id, gain_in_file, peak_count, fc_count = load_single_fits_sampling_interval(single_file)
+            
+            if gain_in_file == gain:
+                if not run_id in self.peak_count_stuck.keys():
+                    self.peak_count_stuck[run_id] = np.zeros([N_PIXELS, N_CAPACITORS_CHANNEL])
+                    self.fc_count_stuck[run_id] = np.zeros([N_PIXELS, N_CAPACITORS_CHANNEL])
+                
+                self.peak_count_stuck[run_id] += peak_count
+                self.fc_count_stuck[run_id] += fc_count
+
+
+    def convert_to_samp_interval_coefficient(self, gain):
+        # convert peak counts to sampling interval coefficient
+
+        for run_id in self.peak_count_stuck.keys():
+            self.sampling_interval_coefficient[run_id] = np.zeros([N_PIXELS, N_CAPACITORS_CHANNEL + N_SAMPLES])
+
+            for pixel in range(N_PIXELS):
+                self.sampling_interval_coefficient[run_id][pixel, :N_CAPACITORS_CHANNEL] = \
+                    self.peak_count_stuck[run_id][pixel] / np.sum(self.peak_count_stuck[run_id][pixel]) * N_CAPACITORS_CHANNEL
+                
+                self.sampling_interval_coefficient[run_id][pixel, N_CAPACITORS_CHANNEL:] = \
+                    self.sampling_interval_coefficient[run_id][pixel, :N_SAMPLES]
+
+                
+    def set_charge_array(self, gain):
+        self.charge_array_before_corr = np.zeros([N_PIXELS, 60000])
+        self.charge_reso_array_before_corr = np.zeros(N_PIXELS)
+
+        for run_id in self.peak_count_stuck.keys():
+            self.charge_array_after_corr[run_id] = np.zeros([N_PIXELS, 60000])
+            self.charge_reso_array_after_corr[run_id] = np.zeros(N_PIXELS)
+            
+
+    def calc_charge(self, count, event, tel_id, gain, r0_r1_calibrator):
+
+        waveform = event.r1.tel[tel_id].waveform
+        
+        first_capacitors = r0_r1_calibrator.first_cap
+        r1_sample_start = r0_r1_calibrator.r1_sample_start.tel[tel_id]
+        r1_sample_end = r0_r1_calibrator.r1_sample_end.tel[tel_id]
+
+        # Check pulse
+        pulse_event_flag = np.sum(np.max(waveform[gain], axis=1) > 100) > 1800
```

Comment:
This needs more context and probably the unnamed constants here should be transformed into either global constants or configurable traitlets.

Reply:
added contexts, and set global constants for them